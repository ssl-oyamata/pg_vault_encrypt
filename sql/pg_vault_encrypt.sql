-- SELECT encrypt('宇宙人になりたい');
SELECT decrypt(encrypt('宇宙人になりたい'));

-- SELECT encrypt('I want to be an alien');
SELECT decrypt(encrypt('I want to be an alien'));

CREATE TABLE test(id int , data cipher);
INSERT INTO test values(1, 'jp');
INSERT INTO test values(2, '宇宙人になりたい');
INSERT INTO test values(3, 'I want to be an alien');
INSERT INTO test values(4, 'I want to be a alien2'::text);
INSERT INTO test values(5, 'I want to be a alien3'::varchar);
INSERT INTO test values(6, 'I want to be a alien4'::char(32));

SELECT * FROM test;

SELECT * FROM test where data = '宇宙人になりたい';
SELECT * FROM test where data LIKE '宇宙人%';
SELECT * FROM test where data LIKE '%になりたい';
SELECT * FROM test where data LIKE '%宇宙人%'::varchar;
SELECT * FROM test where data LIKE '%宇宙まん%';
SELECT * FROM test where data = 'I want to be a alien4'::char(32);
-- SET pg_vault_encrypt.vault_token = "not key";
-- SELECT * FROM test;
-- SET pg_vault_encrypt.vault_token TO DEFAULT ;
--
-- SET pg_vault_encrypt.vault_addr = "http://192.168.0.40:8080";
-- SELECT * FROM test;
-- SET pg_vault_encrypt.vault_addr TO DEFAULT ;
--
-- SET pg_vault_encrypt.vault_transit_engine_path = "dummy";
-- SELECT * FROM test;
-- SET pg_vault_encrypt.vault_transit_engine_path TO DEFAULT ;
--
-- SET pg_vault_encrypt.vault_transit_key_name = "dummy";
-- SELECT * FROM test;
-- SET pg_vault_encrypt.vault_transit_key_name TO DEFAULT ;

DROP TABLE test;