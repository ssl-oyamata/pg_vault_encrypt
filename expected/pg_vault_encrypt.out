-- SELECT encrypt('宇宙人になりたい');
SELECT decrypt(encrypt('宇宙人になりたい'));
     decrypt      
------------------
 宇宙人になりたい
(1 行)

-- SELECT encrypt('I want to be an alien');
SELECT decrypt(encrypt('I want to be an alien'));
        decrypt        
-----------------------
 I want to be an alien
(1 行)

CREATE TABLE test(id int , data cipher);
CREATE TABLE
INSERT INTO test values(1, 'jp');
INSERT 0 1
INSERT INTO test values(2, '宇宙人になりたい');
INSERT 0 1
INSERT INTO test values(3, 'I want to be an alien');
INSERT 0 1
INSERT INTO test values(4, 'I want to be a alien2'::text);
INSERT 0 1
SELECT * FROM test;
 id |         data          
----+-----------------------
  1 | jp
  2 | 宇宙人になりたい
  3 | I want to be an alien
  4 | I want to be a alien2
(4 行)

SELECT * FROM test where data = '宇宙人になりたい';
 id |       data       
----+------------------
  2 | 宇宙人になりたい
(1 行)

SELECT * FROM test where data LIKE '宇宙人%';
 id |       data       
----+------------------
  2 | 宇宙人になりたい
(1 行)

SELECT * FROM test where data LIKE '%になりたい';
 id |       data       
----+------------------
  2 | 宇宙人になりたい
(1 行)

SELECT * FROM test where data LIKE '%宇宙人%';
 id |       data       
----+------------------
  2 | 宇宙人になりたい
(1 行)

SELECT * FROM test where data LIKE '%宇宙まん%';
 id | data 
----+------
(0 行)

-- SET pg_vault_encrypt.vault_token = "not key";
-- SELECT * FROM test;
-- SET pg_vault_encrypt.vault_token TO DEFAULT ;
--
-- SET pg_vault_encrypt.vault_addr = "http://192.168.0.40:8080";
-- SELECT * FROM test;
-- SET pg_vault_encrypt.vault_addr TO DEFAULT ;
--
-- SET pg_vault_encrypt.vault_transit_engine_path = "dummy";
-- SELECT * FROM test;
-- SET pg_vault_encrypt.vault_transit_engine_path TO DEFAULT ;
--
-- SET pg_vault_encrypt.vault_transit_key_name = "dummy";
-- SELECT * FROM test;
-- SET pg_vault_encrypt.vault_transit_key_name TO DEFAULT ;
DROP TABLE test;
DROP TABLE
